kerl

```diff
@@ -1760,10 +1760,20 @@ download() {
 }
 
 github_download() {
+    tarball_file="$KERL_DOWNLOAD_DIR/$1"
+    tarball_url="$OTP_GITHUB_URL/archive/$1"
     # if the file doesn't exist or the file has no size
-    if [ ! -s "$KERL_DOWNLOAD_DIR/$1" ]; then
-        echo "Downloading $1 to $KERL_DOWNLOAD_DIR"
-        curl -f -L -o "$KERL_DOWNLOAD_DIR/$1" "$OTP_GITHUB_URL/archive/$1" || exit 1
+    if [ ! -s $tarball_file ]; then
+        echo "Downloading $1 to $KERL_DOWNLOAD_DIR..."
+        curl -f -L -o $tarball_file $tarball_url || exit 1
+    else
+        # If the downloaded tarball was corrupted due to interruption while
+        # downloading.
+        if ! gunzip -t $tarball_file 2>/dev/null; then
```

Comment:
Shouldn't this be a checksum check? Erlang.org provides archives checksums, no?
Otherwise LGTM.

Reply:
Unfortunately, not possible. The build backend for git doesn't have a checksum file.

```
   327 update_checksum_file() {
   328     if [ "$KERL_BUILD_BACKEND" = 'git' ]; then
   329         return 0
   330     else
   331         echo 'Getting checksum file from erlang.org...'
   332         curl -f -L -o "$KERL_DOWNLOAD_DIR"/MD5 "$ERLANG_DOWNLOAD_URL"/MD5 || exit 1
   333     fi
   334 }
```